geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteFunctionNoAckOp.java

```diff
@@ -236,4 +236,16 @@ protected Message createResponseMessage() {
     }
   }
 
+  private static List<ServerLocation> getAllServers(PoolImpl pool) {
+    List<ServerLocation> servers;
+    if (pool.getLocators() == null || pool.getLocators().isEmpty()) {
+      servers = ((ExplicitConnectionSourceImpl) pool.getConnectionSource()).getAllServers();
```

Comment:
This can be done by introducing a getAllServers() into DataSource interface and modify the AutoConnectionSourceImpl.findAllServers() into getAllServers(). 

Also there's no need to create a static method in ExecuteFunctionNoAckOp.java. Also need to modify ExecuteFunctionOp.java to use ArrayList<ServerLocation> servers = pool.getConnectionSource().getAllServers();